cfu-service/src/splitter.rs

```diff
@@ -0,0 +1,205 @@
+        InternalResponseData::FwVersionResponse(GetFwVersionResponse {
+            header: GetFwVersionResponseHeader::new(1, GetFwVerRespHeaderByte3::NoSpecialFlags),
+            component_info: comp_info,
+        })
+    }
+
+    /// Create an offer rejection response
+    fn create_offer_rejection() -> InternalResponseData {
+        InternalResponseData::OfferResponse(FwUpdateOfferResponse::new_with_failure(
+            HostToken::Driver,
+            OfferRejectReason::InvalidComponent,
+            OfferStatus::Reject,
+        ))
+    }
+
+    /// Create a content rejection response
+    fn create_content_rejection(sequence: u16) -> InternalResponseData {
+        InternalResponseData::ContentResponse(FwUpdateContentResponse::new(
+            sequence,
+            CfuUpdateContentResponseStatus::ErrorInvalid,
+        ))
+    }
+
+    /// Process a fw version request
+    async fn process_get_fw_version(&self) -> InternalResponseData {
+        let mut versions = [FwVersion::default(); MAX_SUPPORTED_DEVICES];
+
+        if self.devices.len() > MAX_SUPPORTED_DEVICES {
+            error!("More devices than supported");
+            return self.create_invalid_fw_version_response();
+        }
+
+        for (device_id, version) in zip(self.devices.iter(), versions.iter_mut()) {
+            if let Ok(InternalResponseData::FwVersionResponse(version_info)) =
+                cfu::route_request(*device_id, RequestData::FwVersionRequest).await
+            {
+                *version = version_info.component_info[0].fw_version;
+            } else {
+                error!("Failed to get FW version for device {}", device_id);
+                return self.create_invalid_fw_version_response();
+            }
+        }
+
+        let overall_version = self.customization.resolve_fw_versions(&versions[..self.devices.len()]);
+        let dev_inf = FwVerComponentInfo::new(overall_version, self.cfu_device.component_id());
+        let comp_info: [FwVerComponentInfo; MAX_CMPT_COUNT] = [dev_inf; MAX_CMPT_COUNT];
+        InternalResponseData::FwVersionResponse(GetFwVersionResponse {
+            header: GetFwVersionResponseHeader::new(1, GetFwVerRespHeaderByte3::NoSpecialFlags),
+            component_info: comp_info,
+        })
+    }
+
+    /// Process a give offer request
+    async fn process_give_offer(&self, offer: &FwUpdateOffer) -> InternalResponseData {
+        let mut offer_responses = [FwUpdateOfferResponse::default(); MAX_SUPPORTED_DEVICES];
+
+        if self.devices.len() > MAX_SUPPORTED_DEVICES {
+            error!("More devices than supported");
+            return Self::create_offer_rejection();
+        }
+
+        for (device_id, offer_response) in zip(self.devices.iter(), offer_responses.iter_mut()) {
+            let mut offer = *offer;
+
+            // Override with the correct component ID for the device
+            offer.component_info.component_id = *device_id;
```

Comment:
Struct update syntax (idk the type names so will need updating)
```suggestion
            let offer = Offer {
                component_info: ComponentInfo {
                    component_id: *device_id,
                    ..offer.component_info
                },
                ..offer
            };
```